### PR TITLE
Change topology format

### PR DESCRIPTION
**Pull request by mpryc**
This proposes new topology format. Instead of using:
topology=N_something, Y_something2, Z_something3

The change proposes:
topology=something:N, something2:Y, something3:Z

### CODE REVIEW

**Comment by rhosqeauto** [comment]
Can one of the admins verify this patch?

**Comment by rhosqeauto** [comment]
Can one of the admins verify this patch?

**Comment by bregman-arie** [comment]
Looks good. Is there a poc job that use this change?

**Comment by mpryc** [comment]
@bregman-arie no, there is no poc job but I've tested the change in the small python code so no regression is introduced and string is properly parsed.

**Comment by yfried-redhat** [comment]
test this please

**Comment by rhosqeauto** [comment]
Can one of the admins verify this patch?

**Comment by mpryc** [comment]
The change was tested on the migi-poc-gate-InfraRed-ospd_qe-openstack-pull-193 job with success.

**Comment by arxcruz** [comment]
Is the old X_something still works? If we introduce this change, all our jobs based on X_something will start to fail, I would like to see compatibility, and set in the documentation that X_something will be deprecated soon.

**Comment by bregman-arie** [comment]
@arxcruz since we have stable version for IR, I think it should be fine merging this to master. Jobs should use stable version.

**Comment by yfried-redhat** [comment]
ok to test

File: cli/spec.py [hunk 1]
```diff
@@ -222,11 +222,21 @@ def resolve_value(self, arg_name, defaults=None):
...
+
+            node_type, number = None, None
+
+            pattern = re.compile("^[A-Za-z]+:[0-9]+$")
             if pattern.match(topology_item) is None:
-                raise exceptions.IRWrongTopologyFormat(self.value)
+                pattern = re.compile("^[0-9]+_[A-Za-z]+$")
+                if pattern.match(topology_item) is None:
+                    raise exceptions.IRWrongTopologyFormat(self.value)
+                number, node_type = topology_item.split('_')
```
Comment (yfried-redhat) [inline cli/spec.py]:
```python
LOG.warn("This topology format is deprecated and will be removed in future versions. Please see `--help` for proper format") 
```

Comment (mpryc) [inline cli/spec.py]: added as LOG.warning

**Comment by arxcruz** [comment]
@bregman-arie not exactly, we have nightly jobs running on master, besides, when master became stable, we need to change all JJB, it can easily broken our gate jobs.

**Comment by mpryc** [comment]
@arxcruz the last fix introduced backwards compability so nothing will be broken.

**Comment by arxcruz** [comment]
@mpryc Awesome! Thanks!

**Comment by yfried-redhat** [comment]
@mpryc Thank you. merged.

**Pull request merged**